plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -627,10 +627,23 @@ public function maybe_update_nonce_user_logged_out( $uid, $action ) {
 	public function cleanup_sessions() {
 		global $wpdb;
 
-		$wpdb->query( $wpdb->prepare( 'DELETE FROM %i WHERE session_expiry < %d', $this->_table, time() ) );
-
-		if ( class_exists( 'WC_Cache_Helper' ) ) {
-			WC_Cache_Helper::invalidate_cache_group( WC_SESSION_CACHE_GROUP );
+		// The initial query can be slow for big session tables, but it doesn't make sense to introduce an index covering session_expiry due to high cardinality.
+		$expired_sessions = $wpdb->get_results( $wpdb->prepare( 'SELECT session_id, session_key FROM %i WHERE session_expiry < %d', $this->_table, time() ) );
+		if ( null !== $expired_sessions ) {
+			$batch_size   = 100;
+			$cache_prefix = $this->get_cache_prefix();
+			// Below, we perform the table and cache cleanup in batches, taking into account the following aspects:
+			// - Deleting smaller batches will lower the sessions table locking time (the cleanup impact on sessions performance will be minimal).
```

Comment:
Isn't the lock an exclusive lock on the row and not the entire DB or table?  I'm wondering if we know the impact of the lock.

I'd like to see the performance impact before we beginning implementing more complex patterns here.